### PR TITLE
Vulkan:Fix crash caused by accessing NULL handle in Vulkan DCE

### PR DESCRIPTION
**Pull request by Qining**
Check if the given Vulkan Handle exist in the state before getting the object of the handle. Sometime using VK_NULL_HANDLE (value 0) is valid, but no object should be obtained from that null handle.

### CODE REVIEW

File: gapis/gfxapi/vulkan/dependency_graph.go [hunk 1]
```diff
@@ -334,12 +334,16 @@ func (g *DependencyGraph) getBehaviour(ctx context.Context, s *gfxapi.State, id
 
 	// Helper function that gets the overlapped memory bindings for a given image
 	getOverlappedBindingsForImage := func(image VkImage) []*vulkanDeviceMemoryBinding {
-		if GetState(s).Images.Get(image).IsSwapchainImage {
+		if !GetState(s).Images.Contains(image) {
+			log.E(ctx, "Error Image: %v: does not exist in state", image)
```
Comment (ben-clayton) [inline gapis/gfxapi/vulkan/dependency_graph.go]: Should this return?

File: gapis/gfxapi/vulkan/dependency_graph.go [hunk 2]
```diff
@@ -349,10 +353,14 @@ func (g *DependencyGraph) getBehaviour(ctx context.Context, s *gfxapi.State, id
 
 	// Helper function that gets the overlapped memory bindings for a given buffer
 	getOverlappedBindingsForBuffer := func(buffer VkBuffer) []*vulkanDeviceMemoryBinding {
-		if GetState(s).Buffers.Get(buffer).Memory != nil {
-			boundMemory := GetState(s).Buffers.Get(buffer).Memory.VulkanHandle
-			offset := uint64(GetState(s).Buffers.Get(buffer).MemoryOffset)
-			size := uint64(uint64(GetState(s).Buffers.Get(buffer).Info.Size))
+		if !GetState(s).Buffers.Contains(buffer) {
+			log.E(ctx, "Error Buffer: %v: does not exist in state", buffer)
```
Comment (ben-clayton) [inline gapis/gfxapi/vulkan/dependency_graph.go]: Likewise, after erroring, is there any point in continuing the function?

File: gapis/gfxapi/vulkan/dependency_graph.go (hunk 1 above)
Comment (Qining) [inline gapis/gfxapi/vulkan/dependency_graph.go]: just added in fa86381

File: gapis/gfxapi/vulkan/dependency_graph.go (hunk 2 above)
Comment (Qining) [inline gapis/gfxapi/vulkan/dependency_graph.go]: Just added in fa86381

**Pull request merged**